### PR TITLE
Fix clippy for object_store

### PR DESCRIPTION
**Pull request by alamb**
More fallout from the most recent rust release

@XiangpengHao  fixed most of the others in https://github.com/apache/arrow-rs/pull/5881

### CODE REVIEW

File: object_store/src/aws/credential.rs [hunk 1]
```diff
@@ -424,6 +424,7 @@ fn canonicalize_headers(header_map: &HeaderMap) -> (String, String) {
 /// <https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/configuring-instance-metadata-service.html>
 #[derive(Debug)]
 pub struct InstanceCredentialProvider {
+    #[allow(dead_code)]
```
Comment (Xuanwo) [inline object_store/src/aws/credential.rs:427]: Would you like to create an issue for this? I feel like we've overlooked something here that prevents `cache` from ever being used. I'm willing to get it fixed.

Comment (alamb) [reply object_store/src/aws/credential.rs:427]: Filed https://github.com/apache/arrow-rs/issues/5884

Comment (alamb) [reply object_store/src/aws/credential.rs:427]:
```suggestion
    // https://github.com/apache/arrow-rs/issues/5884
    #[allow(dead_code)]
```

**Review by Xuanwo — APPROVED** [review]

**Comment by alamb** [comment]
Thank you for the review @Xuanwo

**Pull request merged**